### PR TITLE
Disable auto reload if page is not visible

### PR DESCRIPTION
**Pull request by bmd3k**
* Motivation for features / changes
Reduce the load on TensorBoard servers by disabling auto reload if the page is not considered visible.

* Technical description of changes
Do not auto reload if document is not visible according to the Page
Visibility API. When page again becomes visible perform an auto reload
if one has been missed.

* Detailed steps to verify changes work correctly (as executed by you)
Run a local TensorBoard.
Open localhost:6006.
Ensure option "Reload data every 30s." is selected.
Open detached network tab in chrome developer tools.
Watch for several minutes and observe that many network calls are made every 30s.
Open different tab in same window but with chrome developer tools still visible.
Watch for several minutes and observe that no network calls are made.
Switch back to TensorBoard tab and watch as network calls are immediately made and UI reloads graphs.
Continue to watch network tab as it again follows a 30s period for reload network calls.

### CODE REVIEW

File: tensorboard/components/tf_tensorboard/autoReloadBehavior.ts [hunk 1]
```diff
@@ -59,14 +74,34 @@ namespace tf_tensorboard {
...
+        this._missedAutoReload = true;
       }
-      this.reload();
       this._autoReloadId = window.setTimeout(
         () => this._doAutoReload(),
         this.autoReloadIntervalSecs * 1000
       );
     },
+    _doReload: function() {
+      if (this.reload == null) {
```
Comment (stephanwlee) [inline tensorboard/components/tf_tensorboard/autoReloadBehavior.ts:88]: Prefer triple equals.

File: tensorboard/components/tf_tensorboard/autoReloadBehavior.ts [hunk 2]
```diff
@@ -39,13 +39,28 @@ namespace tf_tensorboard {
...
+      document.addEventListener(
+        'visibilitychange',
+        this._handleVisibilityChange.bind(this)
+      );
+    },
     detached: function() {
       window.clearTimeout(this._autoReloadId);
+      document.removeEventListener(
+        'visibilitychange',
+        this._handleVisibilityChange.bind(this)
```
Comment (stephanwlee) [inline tensorboard/components/tf_tensorboard/autoReloadBehavior.ts:62]: I believe `bind` call creates a new reference and this is not removing the event listener attached at L55.

Comment (bmd3k) [reply tensorboard/components/tf_tensorboard/autoReloadBehavior.ts]: Ya, yuck. I looked at how some other components handle this and picked what looks like the least-bad option.

**Comment by bmd3k** [comment]
> > FWIW, would perhaps be good to update the `Reload data every 30s.` settings panel text to reflect the fact that this is now more complex behavior, e.g. "Reload data every 30s while page is visible" or something?
> > cc @GalOshri
> 
> I am following up with Gal on it. Will send a followup PR if he thinks the message should be updated. My personal opinion is that most people would understand the new behavior without the explanation .

I followed up with Gal. He doesn't think it's necessary to change the existing text.

File: tensorboard/components/tf_tensorboard/autoReloadBehavior.ts (hunk 1 above)
Comment (stephanwlee) [reply tensorboard/components/tf_tensorboard/autoReloadBehavior.ts:94]: ACK

**Pull request merged**